### PR TITLE
Implemented #523 - open function option FunctionOnlyReturningConstant

### PR DESCRIPTION
**Pull request by schalkms**

### CODE REVIEW

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/KtModifierList.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import org.jetbrains.kotlin.psi.KtModifierListOwner
+
+fun KtModifierListOwner.isPublicNotOverridden() =
+		isPublic() && !isOverridden()
+
+fun KtModifierListOwner.isOverridden() = hasModifier(KtTokens.OVERRIDE_KEYWORD)
+
+fun KtModifierListOwner.isOpen() = hasModifier(KtTokens.OPEN_KEYWORD)
+
+fun KtModifierListOwner.isPublic(): Boolean {
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/KtModifierList.kt]: I think we could refactor this to `isPrivate` and where its used negate it?

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/KtModifierList.kt]: I am not quite sure if I understood you correctly.
!isPrivate() is also true for ‘protected’ and ‘internal’

Comment (Mauin) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/KtModifierList.kt]: Agree that `isPublic` is not fully correct as a method name as it checks for much more

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/RedundantVisibilityModifierRule.kt [hunk 2]
```diff
@@ -24,8 +25,7 @@ class RedundantVisibilityModifierRule(config: Config = Config.empty) : Rule(conf
 	private val classVisitor = ClassVisitor()
 	private val childrenVisitor = ChildrenVisitor()
 
-	private fun KtModifierListOwner.isExplicitlyPublicNotOverridden() = isExplicitlyPublic()
-			&& !this.hasModifier(KtTokens.OVERRIDE_KEYWORD)
+	private fun KtModifierListOwner.isExplicitlyPublicNotOverridden() = isExplicitlyPublic() && !isOverridden()
 
 	private fun KtModifierListOwner.isExplicitlyPublic() = this.hasModifier(KtTokens.PUBLIC_KEYWORD)
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/RedundantVisibilityModifierRule.kt]: I'd move this to the new file with extensions as well. This is the true `isPublic` method

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/KtModifierList.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/KtModifierList.kt]: I mean that this method effectively just tests if the modifier is not private. I think it is just used once and if we change the code on that line to !isPrivate, we can delete this method.

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/KtModifierList.kt]: `isPublic()` is used 6 times. 
The method checks if the modifier is not
- private
- protected
- internal

**Pull request merged**